pkgs/development/tools/mbed-cli/default.nix

```diff
@@ -1,22 +1,56 @@
-{ lib, python3Packages, git, mercurial }:
+{ lib, python3Packages, git, mercurial, substituteAll }:
 
 with python3Packages;
 
 buildPythonApplication rec {
   pname = "mbed-cli";
-  version = "1.9.1";
+  version = "1.10.2";
 
   src = fetchPypi {
     inherit pname version;
-    sha256 = "1228plh55id03qywsw0ai88ypdpbh9iz18jfcyhn21pci7mj77fv";
+    sha256 = "0d26rbsyx813i025wmppf020wab1ik20ndi4alnrkp2wzjb59p9n";
   };
 
+  buildInputs = [
+    git
+    mercurial
+  ];
+
+  propagatedBuildInputs = [
+    mbed-os-tools
+    pip
+    pyserial
+
+    # Some of mbed-os dependencies.
```

Comment:
So basically `mbed-cli` tries to run `pip` to install project dependencies.
Essentially all projects using this application will have a dependency on `mbed-os`, and so the following is a list of most of it's dependencies(https://github.com/ARMmbed/mbed-os/blob/master/requirements.txt) - the rest do not have a Nix package.
Not fancy and will probably break if the `mbed-os` dependencies change.

My idea is that users could set up `venv` with the missing dependencies and then use the `mbed-cli` package with the rest.

Any (simple) idea how to handle this better?

Reply:
for packages that require specific dependencies which don't reflect the current python ecosystem you should do something similar to [aws-sam-cli](https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/tools/aws-sam-cli/default.nix)

Two things that are special about it:
 - it's not in python-packages, so it can't be imported
 - it's only used as an application

If you still need to use the package in python-modules, then you will have issues

Reply:
however, for the update story, I don't think that would be a supported usecase as it expects to be able to write to site-packages